assets/src/components/mapMarkers.tsx

```diff
@@ -135,7 +157,19 @@ export const VehicleMarker = ({
         icon={vehicleIcon}
         eventHandlers={eventHandlers}
         zIndexOffset={zIndexOffset}
-      />
+        ref={markerRef}
+      >
+        {inTestGroup(TestGroups.DetoursPilot) && (
+          <Popup className="c-dropdown-popup-wrapper" offset={[140, 97]}>
+            <DropdownMenu>
+              <DropdownItem>
+                Start a detour on route {vehicle.routeId}
+              </DropdownItem>
+            </DropdownMenu>
+          </Popup>
+        )}
```

Comment:
question(non-blocking): do you think we could do this via composistion so it's opt in?
```
<VehicleMarker>
  <DropdownMenu.Popup>
    <DropdownItem>Item</DropdownItem>
  </DropdownMenu.Popup>
</VehicleMarker>
```

I was trying to find a way to do it without having to run a bunch of "wires" but didn't want to spend much more time on it.
(Example of what "wires" I was hoping to avoid. Essentially, it'd be great if `ref` didn't need to be interacted with at all on the top level)

```
<VehicleMarker ref={markerRef} eventHandlers={{click: () => markerRef.current?.closePopup(), contextmenu: () => markerRef.current?.openPopup()}}>
  <DropdownMenu.Popup>
    <DropdownItem>Item</DropdownItem>
  </DropdownMenu.Popup>
</VehicleMarker>
```

Reply:
Hmmmm - I'll play around with this and see if I can come up with anything. I agree that running a bunch of `ref` wires through the interface of `VehicleMarker` will be annoying.

Reply:
I think we can avoid the `ref` wires by keeping `suppressPopup()` where it is and just having the `Popup` itself be passed into `<VehicleMarker />` as a `children`, like you suggested. 

But....

If every vehicle marker should get the popup, then I think I'd rather keep it like this. There are five instances of `<VehicleMarker />` throughout the code, and I'm worried about duplicating the `<DropdownMenu />` call in five places. 

Basically, is the popup an inherent feature of a `<VehicleMarker />`, or is it a feature only of specific `<VehicleMarker />`'s? I _think_ it's the former, but that's more of a design/product question.

Reply:
> If every vehicle marker should get the popup, then I think I'd rather keep it like this. There are five instances of `<VehicleMarker />` throughout the code, and I'm worried about duplicating the `<DropdownMenu />` call in five places.

I don't think every `<VehicleMarker />` is supposed to have it though, AFAIK from the designs, we're only supposed to have it on the Map Page. So we should double check with Mariele on this.

> Basically, is the popup an inherent feature of a `<VehicleMarker />`, or is it a feature only of specific `<VehicleMarker />`'s? I _think_ it's the former, but that's more of a design/product question.

I still think it's better practice to compose these rather than build them together, because if we need to pull them apart in the future, it may get harder if we add more logic in here, but if we build with composition from the start we're offered a lot more flexibility as well as less to test per component and less each component is individually responsible for. If we end up wanting to use them together in more than two locations, building a component which composes them would be my choice. 
I'm less stubborn on this if we do want this _everywhere_ we use vehicle markers, but that is essentially every map right now, including the minimap, which I'm relatively certain the menu is not supposed to be there.

But this is non-blocking, as we _can_ pull these apart later if needed.